test/libsolidity/interface/FileReader.cpp

```diff
@@ -313,8 +313,9 @@ BOOST_AUTO_TEST_CASE(normalizeCLIPathForVFS_should_not_resolve_symlinks_unless_r
 	if (!createSymlinkIfSupportedByFilesystem(tempDir.path() / "abc", tempDir.path() / "sym", true))
 		return;
 
-	boost::filesystem::path expectedPrefixWithSymlinks = "/" / tempDir.path().relative_path();
-	boost::filesystem::path expectedPrefixWithoutSymlinks = "/" / boost::filesystem::weakly_canonical(tempDir).relative_path();
+	boost::filesystem::path expectedRootPath = FileReader::normalizeCLIRootPathForVHS(tempDir);
+	boost::filesystem::path expectedPrefixWithSymlinks = expectedRootPath / tempDir.path().relative_path();
+	boost::filesystem::path expectedPrefixWithoutSymlinks = expectedRootPath / boost::filesystem::weakly_canonical(tempDir).relative_path();
```

Comment:
One thing that's problematic here that after your change we're using the code under test to generate part of the expectations for that same code :)

But I guess, since this particular test is not checking root normalization specifically, it should be fine and is probably less ugly than any alternative.

Reply:
Yes, we partially forge expectations, and I do not like it either. But I hope it is still a reasonable pragmatic fix. The entire topic (starting with [a dangerous `current_path()`](https://www.boost.org/doc/libs/1_68_0/libs/filesystem/doc/reference.html#current_path), is somewhat sensitive and fragile.

> I.e. most of the corner cases really come from Boost 

BTW, why do we use `boost::filesystem` and not `std::filesystem`?

Reply:
Yeah, I think it's acceptable. Solves the problem with temp dir location on Windows so seems worth it to me.

And as much as `current_path()` is dangerous, we have to live with it. Not sure if there's even an alternative.

As for `std::filesystem`, I think it's just it wasn't available when the project started and now we're rolling with it. I'd be fine with either but unfortunately they have slight differences in behavior that would affect normalization (see for example https://github.com/boostorg/filesystem/issues/88#issuecomment-435102621). At least `FileReader` is now well covered so maybe it would not be as risky as it used to be but I can't vouch for anything using `boost::filesystem` outside of that code.